app/Repositories/NeedsRepository.php

```diff
@@ -83,7 +83,7 @@ public function searchNeeds($search){
                     $query = $query->where(DB::Raw('CONCAT_WS(" ",name,telephone,address,city,needs)'),"like","%$searchPart%");
```

Comment:
This query might be open to SQL injection. Since the calls to `searchNeeds` with the parameter is passed to a `DB::raw` which doesn't use prepared statements nor sanitises the input.